src/transaction.ts

```diff
@@ -1966,12 +1973,24 @@ export class Transaction extends Dml {
     const requestOptions = (options as CommitOptions).requestOptions;
     const reqOpts: CommitRequest = {mutations, session, requestOptions};
 
+    const span = getActiveOrNoopSpan();
+
     if (this.id) {
       reqOpts.transactionId = this.id as Uint8Array;
     } else if (!this._useInRunner) {
       reqOpts.singleUseTransaction = this._options;
     } else {
-      this.begin().then(() => this.commit(options, callback), callback);
+      this.begin().then(
```

Comment:
This is not required, as we are calling commit here again which will add `commit failed` event if error.

Reply:
Done, thanks!